src/main/java/org/jenkinsci/plugins/github/config/GitHubServerConfig.java

```diff
@@ -0,0 +1,169 @@
+package org.jenkinsci.plugins.github.config;
+
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.google.common.base.Predicate;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+import org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+
+import static com.cloudbees.plugins.credentials.CredentialsMatchers.firstOrDefault;
+import static com.cloudbees.plugins.credentials.CredentialsMatchers.withId;
+import static com.cloudbees.plugins.credentials.CredentialsProvider.lookupCredentials;
+import static com.cloudbees.plugins.credentials.domains.URIRequirementBuilder.fromUri;
+import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.Validate.notNull;
+
+/**
+ * @author lanwen (Merkushev Kirill)
+ */
+@XStreamAlias("github")
+public class GitHubServerConfig extends AbstractDescribableImpl<GitHubServerConfig> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubServerConfig.class);
+
+    public static final String GITHUB_URL = "https://api.github.com";
+
+    private String apiUrl;
+    private final String credentialsId;
+
+    // only to nullify apiUrl when uncheck
+    private transient boolean enterprise;
+
+    @DataBoundConstructor
+    public GitHubServerConfig(String credentialsId) {
+        this.credentialsId = credentialsId;
+    }
+
+    public String getApiUrl() {
+        return apiUrl;
+    }
+
+    @DataBoundSetter
+    public void setApiUrl(String apiUrl) {
+        if (enterprise) {
+            this.apiUrl = apiUrl;
+        } else {
+            this.apiUrl = null;
+        }
+    }
+
+    @DataBoundSetter
+    public void setEnterprise(boolean enterprise) {
+        this.enterprise = enterprise;
```

Comment:
this is for collapse block with github url (because of only github enterprise has custom url). But if url already defined and you want to uncheck enterprise - so only this method works to nullify apiUrl property.